src/box/alter.cc

```diff
@@ -3367,10 +3367,11 @@ func_def_new_from_tuple(struct tuple *tuple)
 			return NULL;
```

Comment:
The tests fail on this commit. You should ensure that each of your commits passes all the tests. Also as you probably noticed, we usually specify subsystem name in the commit title. For example, `alter: `, `box: `, `raft: `, etc. In your case I could recommend just `box: `. So the title would be `box: show the actual type along with required on field type mismatch`.